### DISCUSSION TITLE
Help Wanted

### DISCUSSION THREAD

**Discussion by basil4240** [FilledStacks/stacked#966] [General]
### Usage of AnimationController or Vsync-requiring Class in Stacked Architecture

Hello everyone,

I am currently working on a project utilizing a stacked architecture in my Flutter application. First and foremost, I would like to express my gratitude to the entire community for developing and maintaining such an amazing library. The framework has been incredibly helpful in building robust and scalable applications.

Recently, I have encountered a challenge while integrating an **AnimationController** or any other class that requires **vsync** into my stacked architecture. I am seeking guidance and assistance on how to properly utilize these classes in this specific architecture.

I'm unsure of the correct approach for incorporating an **AnimationController** or any other class that requires a **vsync** parameter in this architecture. Typically, these classes require a **TickerProvider**, which is usually provided by a **StatefulWidget**. In a stacked architecture, the **ViewModel** does not extend **StatefulWidget**, leading to confusion on how to handle such scenarios.

I would greatly appreciate any advice or guidance on how to correctly integrate **AnimationControllers** or other **vsync**-requiring classes within the stacked architecture. It would be helpful if someone could provide a sample implementation or share their insights on the recommended approach.

Thank you all for your time and support. I look forward to learning from the community's expertise and suggestions.

Best regards,

**Comment by FilledStacks** [comment]
Hey, 

I appreciate your kind words about the project. It makes me really happy to hear that you enjoy it. 

You can use stateful widgets and simply use the `ViewModelBuilder` in your build function. This will allow you to supply the animation controller. 

Another way we do it is be using [StackedHooks](https://github.com/Stacked-Org/hooks) along with the `flutter_hooks` package to create your animation controller.
